stablehlo/conversions/linalg/transforms/StablehloToLinalgRandom.cpp

```diff
@@ -634,12 +634,12 @@ LogicalResult generateLinalgPhilox32(OpBuilder &builder, Location loc,
   // Slice to only the required results.
   collapseShape[0] = resultTy.getNumElements();
 
-  llvm::SmallVector<int64_t> offset(resultTy.getRank(), 0);
-  llvm::SmallVector<int64_t> stride(resultTy.getRank(), 1);
+  llvm::SmallVector<int64_t> offset(collapseShape.size(), 0);
```

Comment:
This is where I made the fix for the index out of bounds issue. See also below in generateLinalgPhilox64.